core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -1742,6 +1746,12 @@ class SocketServerTest {
            selector = Some(testableSelector)
            testableSelector
         }
+
+        override private[network] def processException(errorMessage: String, throwable: Throwable, isUncaught: Boolean): Unit = {
```

Comment:
```isUncaught``` is used by testing only so it is a bit awkward to production code. Could you check the ```errorMessage``` instead of adding new argument? for example:

```scala
if (errorMessage == "Processor got uncaught exception.") uncaughtExceptions += 1
```

Reply:
Funny you should say that, I was initially checking the error message and then felt that the test wouldn't fail if the error message was changed. But since then I had also updated the test which triggers the uncaught exception code path, so it is actually safe now to check the error message. Have updated the code.